osu.Server.BeatmapSubmission/Services/BeatmapPackageParser.cs

```diff
@@ -18,18 +18,36 @@ namespace osu.Server.BeatmapSubmission.Services
 {
     public static class BeatmapPackageParser
     {
+        public static readonly HashSet<string> INVALID_EXTENSIONS = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
```

Comment:
I'm not sure if this is an issue with the zip library we are using in lazer, but I know in stable it was feasible to zip a filename with `../` and potentially escape the destination root. It might be beneficial to add a test covering this (or if that's not possible because it's just not allowed, at least checking that much).

Reply:
That is a very good point, even more so in the various patch endpoints where this becomes an actual problem. Attempted to remedy in https://github.com/ppy/osu-server-beatmap-submission/commit/d663007d42afb923488f6207ff34ac0bf5b1315f; manner of doing so is rather crude, but I couldn't find a better way in like 30 minutes of looking.